synapse/api/auth.py

```diff
@@ -373,19 +334,24 @@ async def _get_appservice_user_id_and_device_id(
                     Codes.EXCLUSIVE,
                 )
 
-        return effective_user_id, effective_device_id, app_service
+        return create_requester(
+            effective_user_id, app_service=app_service, device_id=effective_device_id
+        )
 
     async def get_user_by_access_token(
         self,
         token: str,
         allow_expired: bool = False,
-    ) -> TokenLookupResult:
+        mark_as_used: bool = False,
```

Comment:
I've added this parameter because I moved the `mark_access_token_as_used` here, because the `token_used` exists in the `TokenLookupResult` but not in the `Requester` ; but there are places where we're calling `get_user_by_access_token` and don't really want to mark the token as used in those cases (although I guess we could?)

Reply:
yeah this is fine, though I wonder if the default should be `True` ?

Reply:
It might be better to remove this parameter altogether? (see my comment below)

Reply:
I got rid of this parameter in 1670733e7